make with errors on centos 7.8

$ cmake --version
cmake version 3.18.2
CMake suite maintained and supported by Kitware (kitware.com/cmake).
$ make --version
GNU Make 4.3
Built for x86_64-pc-linux-gnu
Copyright (C) 1988-2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
$ mkdir build && cd build
$ cmake3 ..
-- The CXX compiler identification is GNU 10.2.0
-- Check for working CXX compiler: /opt/gcc/10.2.0/gcc-10.2.0/bin/g++
-- Check for working CXX compiler: /opt/gcc/10.2.0/gcc-10.2.0/bin/g++ - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /TRAVAIL/udpreplay/build
$ make
Scanning dependencies of target udpreplay
[ 50%] Building CXX object CMakeFiles/udpreplay.dir/src/udpreplay.cpp.o
/TRAVAIL/udpreplay/src/udpreplay.cpp: In function 'int main(int, char**)':
/TRAVAIL/udpreplay/src/udpreplay.cpp:225:32: error: 'const struct udphdr' has no member named 'uh_ulen'
  225 |       ssize_t len = ntohs(udp->uh_ulen) - 8;
      |                                ^~~~~~~
/TRAVAIL/udpreplay/src/udpreplay.cpp:232:28: error: 'const struct udphdr' has no member named 'uh_dport'
  232 |       addr.sin_port = udp->uh_dport;
      |                            ^~~~~~~~
make[2]: *** [CMakeFiles/udpreplay.dir/build.make:83: CMakeFiles/udpreplay.dir/src/udpreplay.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:96: CMakeFiles/udpreplay.dir/all] Error 2
make: *** [Makefile:161: all] Error 2

I have the same issue on a redhat7 system..

Can you help me please

--- a/src/udpreplay.cpp
+++ b/src/udpreplay.cpp
@@ -222,14 +222,22 @@ int main(int argc, char *argv[]) {
         }
       }
 
+#ifdef __GLIBC__
+      ssize_t len = ntohs(udp->len) - 8;
+#else
       ssize_t len = ntohs(udp->uh_ulen) - 8;
+#endif
       const u_char *d =
           &p[sizeof(ether_header) + ip->ip_hl * 4 + sizeof(udphdr)];
 
       sockaddr_in addr;
       memset(&addr, 0, sizeof(addr));
       addr.sin_family = AF_INET;
+#ifdef __GLIBC__
+      addr.sin_port = udp->dest;
+#else
       addr.sin_port = udp->uh_dport;
+#endif
       addr.sin_addr = {ip->ip_dst};
       auto n = sendto(fd, d, len, 0, reinterpret_cast<sockaddr *>(&addr),
                       sizeof(addr));

